File: checkpoint0/src/test/java/com/springernature/checkpoint0/GameTests.java
package com.springernature.checkpoint0;

import org.junit.Test;

import static com.springernature.checkpoint0.Cell.O;
import static com.springernature.checkpoint0.Cell.X;
import static org.hamcrest.core.IsEqual.equalTo;
import static org.junit.Assert.assertThat;

public class GameTests {

    @Test
    public void initialStateOfTheGame() {
        Game game = new Game();

        assertThat(game.nextCellToPlace(), equalTo(X));
        assertThat(new BoardRenderer(game.board()).render(), equalTo(
                " | | \n" +
                        "-----\n" +
                        " | | \n" +
                        "-----\n" +
                        " | | \n"
        ));
    }

    @Test
    public void crossMakesTheFirstMove() throws Game.InvalidMove {
        Game game = new Game();
        Game gameAfterMove = game.makeMove(new Move(1, 1), X);

        assertThat(gameAfterMove.nextCellToPlace(), equalTo(O));
        assertThat(new BoardRenderer(gameAfterMove.board()).render(), equalTo(
                " | | \n" +
                        "-----\n" +
                        " |X| \n" +
                        "-----\n" +
                        " | | \n"
        ));
    }

    @Test(expected = Game.InvalidMove.class)
    public void playerMakesInvalidMove() throws Game.InvalidMove {
        Game game = new Game();
        game.makeMove(new Move(100, 100), X);
    }
}

File: checkpoint1/src/test/kotlin/com/springernature/checkpoint1/GameTests.kt
package com.springernature.checkpoint1

import com.springernature.checkpoint1.Cell.O
import com.springernature.checkpoint1.Cell.X
import org.hamcrest.core.IsEqual.equalTo
import org.junit.Assert.assertThat
import org.junit.Test

class GameTests {

    @Test fun `initial state of the game`() {
        val game = Game()

        assertThat(game.nextCellToPlace(), equalTo(X))
        assertThat(BoardRenderer(game.board()).render(), equalTo(
            " | | \n" +
            "-----\n" +
            " | | \n" +
            "-----\n" +
            " | | \n"
        ))
    }

    @Test fun `cross makes the first move`() {
        val game = Game()
        val gameAfterMove = game.makeMove(Move(1, 1), X)

        assertThat(gameAfterMove.nextCellToPlace(), equalTo(O))
        assertThat(BoardRenderer(gameAfterMove.board()).render(), equalTo(
            " | | \n" +
            "-----\n" +
            " |X| \n" +
            "-----\n" +
            " | | \n"
        ))
    }

    @Test(expected = Game.InvalidMove::class)
    fun `player makes invalid move`() {
        val game = Game()
        game.makeMove(Move(100, 100), X)
    }
}